### DISCUSSION TITLE
do you have parameter to remove the prefix 0x ?

### DISCUSSION THREAD

**Discussion by ChainCoderX** [bpierre/addreth#5] [General]
do you have parameter to remove the prefix 0x ?

**Comment by bpierre** [comment]
(I converted this issue to be a discussion)

For the badge, you can pass a function to the [`label` parameter](https://github.com/bpierre/addreth?tab=readme-ov-file#addreth-) to achieve this:

```tsx
<Addreth
  address="0x…"
  label={(address) => address.replace(/^0x/, '')}
/>
```

**Reply by ChainCoderX** [reply]
thank you.

**Reply by ChainCoderX** [reply]
but I got the following errors;

Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server". Or maybe you meant to call this function rather than return it.
  <... address=... actions="none" label={function label}>

**Reply by ChainCoderX** [reply]
I try the explorer function, then get the same error.
explorer={(address) => ({
    name: "Base",
    accountUrl: `https://basescan.com/address/${address}`,
  })}

**Reply by ChainCoderX** [reply]
<img width="827" alt="image" src="https://github.com/bpierre/addreth/assets/49429891/b4779edb-233e-4285-b214-5f29165982df">

I try the  'use server', still failed.

**Reply by ChainCoderX** [reply]
If I use client rending,there will be new problem.
I want to use server rending, Please help me.

**Reply by bpierre** [reply]
Hi @wade-liwei, yes this should be used in a `"use client"` as addreth is using (client) state.

Are you using Next? You probably could split your `"use server"` page in smaller components, and use addreth and other stateful components in separate `"use client"` files.

**Reply by ChainCoderX** [reply]
Yea, I am using Next, 
Could you please give me an example?

**Reply by ChainCoderX** [reply]
I am a backend enginner, frontend beginner.

**Reply by bpierre** [reply]
Yes, the example is based on Next: https://github.com/bpierre/addreth/tree/968cd25eb4947b02a159c050ca12ad6f1c26c0d9/examples/nextjs

**Comment by ChainCoderX** [comment]
very nice,

right now,only have one question,How do I to add a callback when I click the copy button to remove the prefix '0x'.

**Reply by bpierre** [reply]
There is no API to do this, so it would only be possible by modifying this line: https://github.com/bpierre/addreth/blob/968cd25eb4947b02a159c050ca12ad6f1c26c0d9/src/CopyButton.tsx#L47

Could you describe your use case?

**Reply by ChainCoderX** [reply]
got it, thanks.
